.github/workflows/ci.yaml

```diff
@@ -45,6 +45,30 @@ jobs:
           credentials: "${{ secrets.CHARMHUB_TOKEN }}"
           github-token: "${{ secrets.GITHUB_TOKEN }}"
 
+  integration-test-microk8s-backup:
+    name: Integration tests for backups (microk8s)
+    needs:
+      - lib-check
+      - lint
+      - unit-test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup operator environment
+        uses: charmed-kubernetes/actions-operator@main
+        with:
+          provider: microk8s
+          # This is needed until https://bugs.launchpad.net/juju/+bug/1977582 is fixed.
+          bootstrap-options: "--agent-version 2.9.29"
+      - name: Run integration tests
+        run: tox -e backup-integration
+        env:
+          AWS_ACCESS_KEY: "${{ secrets.AWS_ACCESS_KEY }}"
+          AWS_SECRET_KEY: "${{ secrets.AWS_SECRET_KEY }}"
+          GCP_ACCESS_KEY: "${{ secrets.GCP_ACCESS_KEY }}"
+          GCP_SECRET_KEY: "${{ secrets.GCP_SECRET_KEY }}"
```

Comment:
I think @delgod idea was to mock tests with minio, but I am fine to give a try with real S3 providers.

Reply:
I think we need to use the real providers, as was done on https://github.com/canonical/mongodb-operator/blob/1bd0e3cdefa7f2a7fc9a2bef53c1a31a642f061a/tests/integration/backup_tests/test_backups.py#L180. However I also think it's good to test with minio or https://github.com/localstack/localstack to avoid charges related to cloud storage. Let's wait for @delgod to confirm what's the strategy.

Reply:
we will add a test for s3proxy/seaweedfs (juju recommended replacement for minio) when the s3proxy/seaweedfs charm is ready, meanwhile, only once we manually test against minio is enough.

Reply:
I added a fix on 4eec967564fa7993a45b6b294c60d3c0f2991719 and now the charm works with Minio (tested manually).